include/boost/math/distributions/kolmogorov_smirnov.hpp

```diff
@@ -0,0 +1,494 @@
+// libraries (which would be cumbersome), the PDF would present another
+// difficulty. Therefore I am limiting this implementation to the asymptotic
+// form, even though the exact form has trivial values for certain specific
+// values of x and n. For more on trivial values see
+//
+// Ruben H, Gambino J (1982). “The Exact Distribution of Kolmogorov’s Statistic
+// Dn for n ≤ 10.” Annals of the Institute of Statistical Mathematics.
+// 
+// For a good bibliography and overview of the various algorithms, including
+// both exact and asymptotic forms, see
+// https://www.jstatsoft.org/article/view/v039i11
+//
+// As for this implementation: the distribution is parameterized by n (number
+// of observations) in the spirit of chi-squared's degrees of freedom. It then
+// takes a single argument x. In terms of the Kolmogorov-Smirnov statistical
+// test, x represents the distribution of D_n, where D_n is the maximum
+// difference between the CDFs being compared, that is,
+//
+//   D_n = sup|F_n(x) - G(x)|
+//
+// In the exact distribution, x is confined to the support [0, 1], but in this
+// limiting approximation, we allow x to exceed unity (similar to how a normal
+// approximation always spills over any boundaries).
+//
+// As mentioned previously, the CDF is implemented using the \tau
+// parameterization of the fourth Jacobi Theta function as
+//
+// CDF=θ₄(0|2*x*x*n/pi)
+//
+// The PDF is a hand-coded derivative of that function. Actually, there are two
+// (independent) derivatives, as separate code paths are used for "small x"
+// (2*x*x*n < pi) and "large x", mirroring the separate code paths in the
+// Jacobi Theta implementation to achieve fast convergence. Quantiles are
+// computed using a Newton-Raphson iteration from an initial guess that I
+// arrived at by trial and error.
+//
+// The mean and variance are implemented using simple closed-form expressions.
+// The mode is calculated at run-time by maximizing the PDF. If you have an
+// analytical solution for the mode, feel free to plop it in.
+//
+// The CDF and PDF could almost certainly be re-implemented and sped up using a
+// polynomial or rational approximation, since the only meaningful argument is
+// x * sqrt(n). But that is left as an exercise for the next maintainer.
+//
+// In the future, the Pelz-Good approximation could be added. I suggest adding
+// a second parameter representing the order, e.g.
+//
+// kolmogorov_smirnov_dist<>(100) // N=100, order=1
+// kolmogorov_smirnov_dist<>(100, 1) // N=100, order=1, i.e. Kolmogorov's formula
+// kolmogorov_smirnov_dist<>(100, 4) // N=100, order=4, i.e. Pelz-Good formula
+//
+// The exact distribution could be added to the API with a special order
+// parameter (e.g. 0 or infinity), or a separate distribution type altogether
+// (e.g. kolmogorov_smirnov_exact_distribution).
+//
+#ifndef BOOST_STATS_KOLMOGOROV_ASYMPTOTIC_HPP
+#define BOOST_STATS_KOLMOGOROV_ASYMPTOTIC_HPP
```

Comment:
I think this follows a pattern which was inherited from some very old code. I'd use `#ifndef BOOST_MATH_DISTRIBUTIONS_KOLMOGOROV_SMIRNOV_HPP` to match our current practice.

Reply:
Got it, done